src/ui/run-info-ui-handler.ts

```diff
@@ -0,0 +1,731 @@
+      const heldItemsContainer = this.scene.add.container(-82, 6);
+      const heldItemsList : PokemonHeldItemModifier[] = [];
+      if (this.runInfo.modifiers.length) {
+        for (const m of this.runInfo.modifiers) {
+          const modifier = m.toModifier(this.scene, this.modifiersModule[m.className]);
+          if (modifier instanceof PokemonHeldItemModifier && modifier.pokemonId === pokemon.id) {
+            modifier.stackCount = m["stackCount"];
+            heldItemsList.push(modifier);
+          }
+        }
+        if (heldItemsList.length > 0) {
+          (heldItemsList as PokemonHeldItemModifier[]).sort(modifierSortFunc);
+          let row = 0;
+          for (const [index, item] of heldItemsList.entries()) {
+            if ( index > 36 ) {
+              const overflowIcon = addTextObject(this.scene, 182, 4, "+", TextStyle.WINDOW);
+              heldItemsContainer.add(overflowIcon);
+              break;
+            }
+            const itemIcon = item?.getIcon(this.scene, true);
+            //itemIcon.setFrame(item.type.iconImage);
+            itemIcon.setScale(heldItemsScale);
+            itemIcon.setPosition((index%19)*10, row*10);
+            heldItemsContainer.add(itemIcon);
+            if (index !== 0 && index%18 === 0) {
+              row++;
+            }
+          }
+        }
+      }
+      heldItemsContainer.setName("heldItems");
+      heldItemsContainer.setVisible(false);
+
+      pokemonInfoContainer.add(pokemonInfoWindow);
+      iconContainer.add(icon);
+      pokemonInfoContainer.add(iconContainer);
+      marksContainer.setName("PkmnMarks");
+      pokemonInfoContainer.add(marksContainer);
+      movesetContainer.setName("PkmnMoves");
+      pokemonInfoContainer.add(movesetContainer);
+      pokeInfoTextContainer.setName("PkmnInfoText");
+      pokemonInfoContainer.add(pokeInfoTextContainer);
+      pokeStatTextContainer.setName("PkmnStatsText");
+      pokemonInfoContainer.add(pokeStatTextContainer);
+      pokemonInfoContainer.add(heldItemsContainer);
+      pokemonInfoContainer.setName("PkmnInfo");
+      this.partyContainer.add(pokemonInfoContainer);
+      pokemon.destroy();
+ 		});
+    this.gameStatsContainer.add(this.partyContainer);
+ 	}
+
+  showParty(partyVisible: boolean): void {
+    const allContainers = this.partyContainer.getAll("name", "PkmnInfo");
+    allContainers.forEach((c: Phaser.GameObjects.Container) => {
+      (c.getByName("PkmnMoves") as Phaser.GameObjects.Container).setVisible(partyVisible);
+      (c.getByName("PkmnInfoText") as Phaser.GameObjects.Container).setVisible(partyVisible);
+      (c.getByName("PkmnStatsText") as Phaser.GameObjects.Container).setVisible(partyVisible);
+      (c.getByName("PkmnMarks") as Phaser.GameObjects.Container).setVisible(partyVisible);
+      (c.getByName("heldItems") as Phaser.GameObjects.Container).setVisible(!partyVisible);
```

Comment:
If you look at the definition of `getByName()`:
```ts
getByName<T extends Phaser.GameObjects.GameObject>(name: string): T;
```
you can notice that passing custom interface as return type is possible through generic type definition.
You should use that

```suggestion
      c.getByName<Phaser.GameObjects.Container>("PkmnMoves").setVisible(partyVisible);
      c.getByName<Phaser.GameObjects.Container>("PkmnInfoText").setVisible(partyVisible);
      c.getByName<Phaser.GameObjects.Container>("PkmnStatsText").setVisible(partyVisible);
      c.getByName<Phaser.GameObjects.Container>("PkmnMarks").setVisible(partyVisible);
      c.getByName<Phaser.GameObjects.Container>("heldItems").setVisible(!partyVisible);
```